keanu-python/keanu/vertex/base.py

```diff
@@ -202,6 +211,94 @@ class Integer(Vertex):
+            "floor_divide": Integer.__rfloordiv__,
+        }
+        if method == "__call__":
+            try:
+                dispatch_method = methods[ufunc.__name__]
+                result = dispatch_method(input1, input0)  # type: ignore # Does not accept Vertex as an Integer
+                return result
+            except KeyError:
+                return super().__array_ufunc__(ufunc, method, input0, input1)
+        else:
+            raise NotImplementedError("NumPy ufunc method %s not implemented" % method)
+
+    def __add__(self, other: vertex_operation_param_types) -> 'Vertex':
+        if is_floating_type(other):
+            return kn.vertex.generated.Addition(kn.vertex.generated.CastToDouble(self), other)
+
+        return kn.vertex.generated.IntegerAddition(self, other)
+
+    def __radd__(self, other: vertex_operation_param_types) -> 'Vertex':
+        if is_floating_type(other):
+            return kn.vertex.generated.Addition(kn.vertex.generated.CastToDouble(self), other)
+
+        return kn.vertex.generated.IntegerAddition(other, self)
+
+    def __sub__(self, other: vertex_operation_param_types) -> 'Vertex':
+        if is_floating_type(other):
+            return kn.vertex.generated.Difference(kn.vertex.generated.CastToDouble(self), other)
+
+        return kn.vertex.generated.IntegerDifference(self, other)
+
+    def __rsub__(self, other: vertex_operation_param_types) -> 'Vertex':
+        if is_floating_type(other):
+            return kn.vertex.generated.Difference(other, kn.vertex.generated.CastToDouble(self))
+
+        return kn.vertex.generated.IntegerDifference(other, self)
+
+    def __mul__(self, other: vertex_operation_param_types) -> 'Vertex':
+        if is_floating_type(other):
+            return kn.vertex.generated.Multiplication(kn.vertex.generated.CastToDouble(self), other)
+
+        return kn.vertex.generated.IntegerMultiplication(self, other)
+
+    def __rmul__(self, other: vertex_operation_param_types) -> 'Vertex':
+        if is_floating_type(other):
+            return kn.vertex.generated.Multiplication(other, kn.vertex.generated.CastToDouble(self))
+
+        return kn.vertex.generated.IntegerMultiplication(other, self)
+
+    def __pow__(self, other: vertex_operation_param_types) -> 'Vertex':
+        if is_floating_type(other):
+            return kn.vertex.generated.Power(kn.vertex.generated.CastToDouble(self), other)
+
+        return kn.vertex.generated.IntegerPower(self, other)
+
+    def __rpow__(self, other: vertex_operation_param_types) -> 'Vertex':
+        if is_floating_type(other):
+            return kn.vertex.generated.Power(other, kn.vertex.generated.CastToDouble(self))
+
+        return kn.vertex.generated.IntegerPower(other, self)
+
+    def __truediv__(self, other: vertex_operation_param_types) -> 'Vertex':
+        other = cast_to_double_vertex_if_integer_vertex(other)
+        return kn.vertex.generated.Division(kn.vertex.generated.CastToDouble(self), other)
```

Comment:
I would find this logic easier to understand:

    return kn.vertex.generated.CastToDouble(self).__truediv__(other)

similarly for `rtruediv`.

Reply:
Yeah that's nice. I can do that in pretty much all of the integer ones so have done that.